relay-redis/src/real.rs

```diff
@@ -382,3 +396,119 @@ pub struct Stats {
     /// The number of idle connections.
     pub idle_connections: u32,
 }
+
+/// [`bb8::ManageConnection`] for a async redis cluster.
+pub struct RedisClusterConnectionManager {
+    client: ClusterClient,
+}
+
+impl RedisClusterConnectionManager {
+    /// Creates a new [`bb8::ManageConnection`] with the provided [`ClusterClient`]
+    pub fn new(client: ClusterClient) -> Self {
+        Self { client }
+    }
+}
+
+impl bb8::ManageConnection for RedisClusterConnectionManager {
+    type Connection = ClusterConnection;
+    type Error = redis::RedisError;
+
+    fn connect<'life0, 'async_trait>(
+        &'life0 self,
+    ) -> Pin<Box<dyn Future<Output = Result<Self::Connection, Self::Error>> + Send + 'async_trait>>
+    where
+        'life0: 'async_trait,
+    {
+        let client = self.client.clone();
+        Box::pin(async move { client.get_async_connection().await })
+    }
+
+    fn is_valid<'life0, 'life1, 'async_trait>(
+        &'life0 self,
+        conn: &'life1 mut Self::Connection,
+    ) -> Pin<Box<dyn Future<Output = Result<(), Self::Error>> + Send + 'async_trait>>
+    where
+        'life0: 'async_trait,
+        'life1: 'async_trait,
+    {
+        Box::pin(async move { redis::cmd("PING").query_async(conn).await })
+    }
+
+    fn has_broken(&self, _: &mut Self::Connection) -> bool {
+        false
+    }
+}
+
+/// An abstraction over Single and Cluster redis connections.
+#[derive(Debug, Clone)]
+pub enum AsyncRedisPool {
+    /// Represents a connection pool to a clustered redis instance.
+    Cluster(bb8::Pool<RedisClusterConnectionManager>, RedisConfigOptions),
+    /// Represents a connection pool to a single redis server.
+    Single(
+        bb8::Pool<bb8_redis::RedisConnectionManager>,
+        RedisConfigOptions,
+    ),
+}
+
+impl AsyncRedisPool {
+    /// Takes a command and executes it on redis.
+    pub async fn query_async<T: FromRedisValue>(&self, cmd: Cmd) -> Result<T, RedisError> {
```

Comment:
I would just call this `query`.

Reply:
The `redis` crate also uses this naming schema so for consistency reasons I adapted the naming